core/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -213,6 +213,38 @@ protected static Object convertToStringIfBytesRef(Object obj) {
     }
 
     /**
+     * Same as {@link #convertToBytesRefIfString} but on Iterable.
+     * @param objs the Iterable of input object
+     * @return the same input or a list of {@link BytesRef} representation if input was a list of type string
+     */
+    protected static List<Object> convertToBytesRefListIfStringList(Iterable objs) {
+        if (objs == null) {
+            return null;
+        }
+        List<Object> newObjs = new ArrayList<>();
+        for (Object obj : objs) {
+            newObjs.add(convertToBytesRefIfString(obj));
+        }
+        return newObjs;
+    }
+
+    /**
+     * Same as {@link #convertToStringIfBytesRef} but on Iterable.
+     * @param objs the Iterable of input object
+     * @return the same input or a list of utf8 string if input was a list of type {@link BytesRef}
+     */
+    protected static List<Object> convertToStringListIfBytesRefList(Iterable objs) {
```

Comment:
not a big fan of these two new methods to be honest. Maybe since they are needed in one query only we can loop there instead?